tests/test_nn.py

```diff
@@ -18,7 +19,7 @@ def test_direct_shape(J, Q, T, padding_mode):
         padding_mode=padding_mode,
     )
     y = graph(x)
-    assert y.shape == (B, C, Q, J, 2**4)
+    assert y.shape[:4] == (B, C, Q, J)
```

Comment:
?

Reply:
Because of the padding effect, the length of y may be greater to $N/2^J$, I can't figure out a mathematical formula for the length of output, (which may depend on the filter length, the input data length, and also the decomposition level J), but y can still be passed to the second layer. Is it necessary to make it exactly equal to $N/2^J$?

Reply:
right now you seem to make it sound like the PyTorch array `y` may have more than four dimensions, and regardless of `N` and `J` i don't see how that can happen

Reply:
i understand now, the fifth dimension is `N` and depends on the input signal